docs/react-native-windows-init.md

```diff
@@ -0,0 +1,66 @@
+# React native windows initialization
+
+This document describes how the template initialization works for `react-native-windows init`.
+
+## Where is the code:
+The main logic that determines the template values and which files to apply is function [`copyProjecTemplateAndReplace` in `vnext/local-cli/generator-windows/index.js`](https://github.com/microsoft/react-native-windows/blob/master/vnext/local-cli/generator-windows/index.js#L48).
+The parameters for `init` is located in variable [`argv` in `packages/react-native-windows-init/src/Cli.ts`](https://github.com/microsoft/react-native-windows/blob/7de1f3843a9754c7fbf744bf3e556fdca78472c1/packages/react-native-windows-init/src/Cli.ts#L27)
+
+The template files can be found in [`vnext/local-cli/generator-windows/templates`](https://github.com/microsoft/react-native-windows/tree/7de1f3843a9754c7fbf744bf3e556fdca78472c1/vnext/local-cli/generator-windows/templates)
+
+## Debugging:
+To debug the template [Contributing to the CLI](https://github.com/microsoft/react-native-windows/wiki/Contributing-to-the-CLI) on our wiki.
+
+## Templating tech (Mustache)
+PR [#5126](https://github.com/microsoft/react-native-windows/pull/5126) updated the template generation for react-native-windows from regex replace to use the mustache template library. 
+### Benefits
+This has the following benefits:
+* We can use conditionals and foreach in the template allowing for better expersivity
+* This expresivity shows that we don't have to construct Xml fragments in Javascript anymore to be inserted by regex replace
```

Comment:
* This expressivity shows that we don't have to construct XML fragments in JavaScript anymore to be inserted by `regex.replace`